### PR TITLE
[CARBONDATA-734] Support the syntax of 'STORED BY PARQUET/ORC' in CarbonSession

### PR DESCRIPTION
**Pull request by watermen**
When I create parquet table with the command below:
```shell
import org.apache.spark.sql.SparkSession 
import org.apache.spark.sql.CarbonSession._
val carbon = SparkSession.builder().config(sc.getConf).getOrCreateCarbonSession("hdfs://100.0.0.4:9000/user/hive/warehouse/carbon.store")
carbon.sql("create table src(key int, value string) stored as parquet")
```

It will failed with the error info below:
```shell
java.lang.RuntimeException: [1.1] failure: identifier matching regex (?i)ALTER expected

create table src(key int, value string) stored as parquet
^
  at scala.sys.package$.error(package.scala:27)
  at org.apache.spark.sql.parser.CarbonSpark2SqlParser.parse(CarbonSpark2SqlParser.scala:45)
  at org.apache.spark.sql.parser.CarbonSparkSqlParser.parsePlan(CarbonSparkSqlParser.scala:51)
  at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:592)
  ... 50 elided
```

The reason is if we use `STORED AS`, the value of `value.storageHandler()` will be null(because it is a fileFormat, not storageHandler), so if we still invoke `STRING()`, it will throw NPE.

cc @jackylk

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
add to whitelist

**Comment by ravipesala** [comment]
@watermen  Thanks for working on this. Please add one testcase inside spark2 package.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1002/

**Comment by watermen** [comment]
@ravipesala Please review the testcase.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1018/

File: integration/spark2/src/test/scala/org/apache/spark/SparkCommandSuite.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark
+
+import org.apache.spark.sql.common.util.QueryTest
+import org.scalatest.BeforeAndAfterAll
+
+class SparkCommandSuite extends QueryTest with BeforeAndAfterAll {
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/spark/SparkCommandSuite.scala]: it is better to add this testcase in `CarbonCommandSuite`, they all use CarbonSession

Comment (watermen) [inline integration/spark2/src/test/scala/org/apache/spark/SparkCommandSuite.scala]: @jackylk But it's not carbon command, So I think maybe we should add spark/hive command in SparkCommandSuite(Also use CarbonSession).

Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/spark/SparkCommandSuite.scala]: ok

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**